Sources/Tracing/Docs.docc/Guides/InstrumentYourLibrary.md

```diff
@@ -0,0 +1,345 @@
+
+```swift
+var span: any Span
+
+func onError(error: Error) {
+  span.recordError(error) // record the error, and...
+  span.end() // end the span
+}
+```
+
+It is worth noting that double-ending a span should be considered a programmer error, and tracer implementations are free to crash or otherwise report such problem. 
+
+>  Note: The problem with finding a span that was ended in two places is that its lifecycle seems to be incorrectly managed, and therefore the span timing information is at risk of being incorrect.
+>
+> Please also take care to never `end()` a span that was created using `withSpan()`  APIs, because `withSpan` will automatically end the span when the closure returns.
+
+#### Storing and restoring baggage across callbacks 
+
+Note also since a `Span` contains an instrumentation `Baggage`, you can also pass the span's baggage to any APIs which may need it, or even restore the baggage e.g. for loggers to pick it up while emitting log statements:
+
+```swift
+final class StatefulHandler {
+    var span: any Span
+
+    func startHandling(request: HTTPRequest) {
+        self.span = InstrumentationSystem.tracer.startSpan("\(request.path)")
+    }
+
+    // callback, form other task, so we don't have the task-local information here anymore
+    func onSomethingHappening(event: SomeEvent) {
+        Baggage.withValue(span.baggage) { // restore task-local baggage
+            // which allows the baggage to be used by loggers and tracers as usual again: 
+            log.info("Event happened: \(event)")
+
+            // since the baggage was restored here, the startSpan will pick it up,
+            // and the "event-id" span will be a child of the "request.path" span we started before.
+            withSpan("event-\(event.id)") { span in // new child span (child of self.span)
+                // ... handle the event ...
+            }
+        }
+    }
+}
+```
+
+It is also possible to pass the baggage explicitly to `withSpan` or `startSpan`:
+
+```swift
+withSpan("event-\(event.id)", baggage: span.baggage) { span in
+  // ... 
+}
+```
+
+which is equivalent to surrounding thr withSpan with a binding of the baggage. The passed baggage (with the values updated by the tracer), will then be set for the duration of the `withSpan` operation, just like usual.
+
+### Global vs. "Stored" Tracers and Instruments
+
+Tracing functions similarly to swift-log and swift-metrics, in the sense that there is a global "backend" configured at application start, by end-users (developers) of an application. And this is how using `InstrumentationSystem/tracer` gets the "right" tracer at runtime.
```

Comment:
missing word here?

```suggestion
Tracing functions work similarly to swift-log and swift-metrics, in the sense that there is a global "backend" configured at application start, by end-users (developers) of an application. And this is how using `InstrumentationSystem/tracer` gets the "right" tracer at runtime.
```

Reply:
The way I read it here is that "it functions" is a substitute for "it works" instead of referring to functions as in methods.

Reply:
Right yeap it's as a verb here.

Reply:
Maybe better to rephrase as "Tracing works" to avoid this potential ambiguity?

Reply:
Yeah that's fair -- will do 👍